lib/neovim-plugin.nix

```diff
@@ -99,21 +102,36 @@ with lib;
             example = settingsExample;
           };
         }
+        // optionalAttrs allowLazyLoad {
+          lazyLoad = nixvimOptions.mkLazyLoadOption {
+            inherit originalName cfg;
+            optionsForPlugin = true;
+            lazyLoad = {
+              # TODO: extract extraConfigLua and extraConfigLuaPre from extraConfig
```

Comment:
IDK if this is a practical goal, due to the way the module system works.

In theory, `extraConfig` functions could be returning a `mkIf`, `mkDefault`, `mkMerge`, etc assuming that the module system will be able to merge it.

It's also possible that plugins define some of their config in other imported modules, e.g. cmp does this, as does none-ls, and several others.

Reply:
One way to handle this would be to add internal options to each plugin of the form `<pluginName>Config(Pre|Post)?`, and have the mkNeovimPlugin put the code in the correct places according to the specified options

Reply:
> One way to handle this would be to add internal options to each plugin of the form `<pluginName>Config(Pre|Post)?`, and have the mkNeovimPlugin put the code in the correct places according to the specified options

IMO having `plugins.*.{pre,post,}config` options would be useful, not just internally but to users too.

Updating the codebase to actually make use of such options internally would be a significant chore though.

Reply:
You are right, this adds some complexity that may be out of scope for now.